beacon_node/beacon_chain/src/migrate.rs

```diff
@@ -49,18 +48,23 @@ pub trait Migrate<S: Store<E>, E: EthSpec>: Send + Sync + 'static {
 
         // Collect hashes from new_finalized_block back to old_finalized_block (inclusive)
         let mut found_block = false; // hack for `take_until`
-        let newly_finalized_blocks: HashMap<SignedBeaconBlockHash, Slot> = HashMap::from_iter(
+        let maybe_newly_finalized_blocks: Result<HashMap<SignedBeaconBlockHash, Slot>, Error> =
             ParentRootBlockIterator::new(&*store, new_finalized_block_hash.into())
-                .take_while(|(block_hash, _)| {
-                    if found_block {
-                        false
-                    } else {
-                        found_block |= *block_hash == old_finalized_block_hash.into();
-                        true
+                .take_while(|result| match result {
+                    Ok((block_hash, _)) => {
+                        if found_block {
+                            false
+                        } else {
+                            found_block |= *block_hash == old_finalized_block_hash.into();
+                            true
+                        }
                     }
+                    Err(_) => true,
                 })
-                .map(|(block_hash, block)| (block_hash.into(), block.slot())),
-        );
+                .map(|result| result.map(|(block_hash, block)| (block_hash.into(), block.slot())))
+                .collect();
+
+        let newly_finalized_blocks = maybe_newly_finalized_blocks?;
```

Comment:
This is just a (subjective) style thing, but I prefer not to have intermediate vars named, only to have them immediately discarded. How do you feel about using `.collect::<Result<_, _>>()?` instead?

Reply:
Yep, it's cleaner 👍